src/Action.php

```diff
@@ -60,6 +60,28 @@ public function __construct(BaseAction &$action)
     }
 
     /**
+     * Returns the path for an action source file.
+     *
+     * @param string $filename
+     * @return string
+     * @throws \Exception
+     */
+    private function getSourcePath(string $filename): string
+    {
+        $path = implode(DIRECTORY_SEPARATOR, [
+            dirname($_SERVER['SCRIPT_NAME']),
+            'actions',
+            "$filename.php"
+        ]);
+
+        if (!is_readable($path)) {
+            throw new \Exception('Source file not readable');
```

Comment:
Instead of throwing an `Exception` this should really use the `error()` function to log an error.